.github/workflows/check-pod.yaml

```diff
@@ -40,3 +40,8 @@ jobs:
           unzip -l Ably.framework.zip | grep 'iOS/Ably.framework'
       - name: Validate pod
         run: pod lib lint
+      - name: Archive built framework
+        uses: actions/upload-artifact@v3
+        with:
+          name: carthage-built-framework
+          path: Ably.framework.zip
```

Comment:
I downloaded it, and during unzip this happens - it produces `Ably.framework.zip` and immediately unzip it further into Carthage folder:

https://user-images.githubusercontent.com/45491869/199824768-4d8abd49-aabb-4718-9b6d-450645742eb3.mov

Reply:
I noticed that, and you're right, it's not good. I should have done something about it in the first place. Will see what I can do.

Reply:
I think it's macos sees that it's an embed archive and continues to unzip it. Probably you should include a command that unzips it and zips again with a proper name.

Reply:
I've introduced a directory at the top level of the artifact zip file; this means that Archive Utility won't try to continue unzipping. (Explained it in the action YAML file too.)

Reply:
Hopefully this is how it'll behave now:

https://user-images.githubusercontent.com/53756884/199828198-63874010-2465-413b-8b09-ee05fb674180.mov